### ISSUE TITLE
Check DB Connection status

### ISSUE THREAD

**Issue by bubnenkoff** [agilord/postgres_pool#10]
Is there any way to check `isConnected` status, like it done in non-pool driver?

**Comment by isoos** [comment]
Which `isConnected` method are you referring to?
What's the use case it would fit? I'd assume that you use the pool while in an active state, and then close it...

**Comment by bubnenkoff** [comment]
Oh sorry, I correct name is `isClosed`, not `isOpen`.
I am writing service that use connection pool, and I want to check it's status. Like uptime, and db status, so only way it to check something like `isConnected`/`isClosed`.

Should I really close connections pool? My use-case that service can be idle for few hours, but in one time it can get multiple sql-insert statements that need pooling.

**Comment by isoos** [comment]
I'm not an expert in the depths of a TCP connection, but I think we can't really tell if a connection is really alive, only if we had traffic on it in the past N seconds. `isClosed` is kind of special, because we know from our side that we expect no more traffic, and if there were traffic, we would drop it.

> Should I really close connections pool? My use-case that service can be idle for few hours, but in one time it can get multiple sql-insert statements that need pooling.

If you look at the default values, it may serve you well:
https://github.com/agilord/postgres_pool/blob/master/lib/postgres_pool.dart#L244-L254

E.g. after a few hour of idle time, if you start to use it for a new query, it will either open a new one (after 12 hours of total living time), or start with a connection tester query first, and then execute your queries...

If you need some metrics that you'd like to expose, I'm happy to help with that, but on the pool level, it may not be the same as on the individual connection level...

**Comment by bubnenkoff** [comment]
>If you need some metrics that you'd like to expose, I'm happy to help with that,

I am doing micro-server architecture, and doing monitor-app that should check if they are alive. I would like also to check if they connection to DB is ok.

Do you have any ideas about it?

**Comment by isoos** [comment]
>  check if they are alive. I would like also to check if they connection to DB is ok.

If you define the condition to be alive as "Had traffic in the past 20-300 seconds", then it would be easy to expose that. (You can also force `SELECT 1;` statements every X minutes if you really want to have them around.)

**Comment by bubnenkoff** [comment]
Yeah, thanks, I also thought about it, it's seems to be good idea. I will think about it a little bit more, and will close issue soon.
